apps/api/src/app/auth/services/passport/jwt.strategy.ts

```diff
@@ -1,25 +1,30 @@
+import type http from 'http';
 import { ExtractJwt, Strategy } from 'passport-jwt';
 import { PassportStrategy } from '@nestjs/passport';
 import { Injectable, UnauthorizedException } from '@nestjs/common';
 import { UserSessionData } from '@novu/shared';
 import { AuthService, Instrument } from '@novu/application-generic';
+import { HttpRequestHeaderKeysEnum } from '../../../shared/framework/types';
 
 @Injectable()
 export class JwtStrategy extends PassportStrategy(Strategy) {
   constructor(private readonly authService: AuthService) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
       secretOrKey: process.env.JWT_SECRET,
+      passReqToCallback: true,
     });
   }
 
   @Instrument()
-  async validate(payload: UserSessionData) {
+  async validate(req: http.IncomingMessage, payload: UserSessionData) {
     const user = await this.authService.validateUser(payload);
     if (!user) {
       throw new UnauthorizedException();
     }
 
-    return payload;
+    const environmentId = req.headers[HttpRequestHeaderKeysEnum.NOVU_ENVIRONMENT_ID.toLowerCase()];
```

Comment:
Read environment id from the corresponding header

Reply:
We should throw an error incase the environmentId is missing here

Reply:
Done.

Reply:
If not mistaken the EnvironmentCommand requires the environment ID using class validator. If not provided, a 400 will be raised for all controllers that need it. 

[See here](https://github.com/novuhq/novu/pull/5768/files#diff-1b0e3f7797aad06baa5dea4639298a417be5d4b075bb84017056e6430c627b9fR6)

Moreover, we can't throw an error at such a centralized place, as not all API routes require environment ID. That is, to add it here, we need a list of endpoints that need it or not needed and act accordingly. But that's a maintenance headache.